implants/lib/eldritch/src/file/list_impl.rs

```diff
@@ -30,40 +32,36 @@ fn windows_tick_to_unix_tick(windows_tick: u64) -> i64 {
     return (windows_tick / WINDOWS_TICK - SEC_TO_UNIX_EPOCH) as i64;
 }
 
-fn create_file_from_dir_entry(dir_entry: DirEntry) -> Result<File> {
-    let mut sys = System::new();
+fn create_file_from_pathbuf(path_entry: PathBuf) -> Result<File> {
+    let mut sys: System = System::new();
     sys.refresh_users_list();
+    let file_name = path_entry
+        .file_name()
+        .context("file.list: Failed to get filename")?
+        .to_str()
+        .context("file.list: Unable to convert file name to string")?
+        .to_string();
 
-    let file_name = match dir_entry.file_name().into_string() {
-        Ok(local_file_name) => local_file_name,
-        Err(_) => {
-            return Err(anyhow::anyhow!(
-                "file.list: Unable to convert file name to string."
-            ))
-        }
-    };
+    let absolute_path = fs::canonicalize(&path_entry)?
+        .to_str()
+        .context("file.list: Failed to canonicalize path")?
+        .to_string();
 
-    let file_type = match dir_entry.file_type() {
-        Ok(tmp_file_type) => {
-            if tmp_file_type.is_dir() {
-                FileType::Directory
-            } else if tmp_file_type.is_file() {
-                FileType::File
-            } else if tmp_file_type.is_symlink() {
-                FileType::Link
-            } else {
-                FileType::Unknown
-            }
-        }
-        Err(_) => FileType::Unknown,
+    let file_type = if path_entry.is_dir() {
```

Comment:
Perhaps we can `match` using [guards](https://doc.rust-lang.org/rust-by-example/flow_control/match/guard.html):

```rust
match path_entry {
    FileType::Directory(t) if path_entry.is_dir() => t,
    FileType::File(t) if path_entry.is_file() => t,
    FileType::Link(t) if path_entry.is_symlink() => t,
    _ => FileType::Unknown,
}
```